portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,28 +31,63 @@
 import com.google.gson.Gson;
 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> messages;
-  
-  @Override 
-  public void init() {
-    messages = new ArrayList<String>(); 
-    messages.add("Hello, this is my first comment");
-    messages.add("Hello, this is my second comment");
-    messages.add("Hi, this is my third comment");
-  }
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private String name;
+  private String category;
+  private String content;
+  private String comment;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String jsonResponse = convertToJson(messages);
+    Query query = new Query("Recommendation").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    //Number limit for entities loaded
+    int maxNumber = results.countEntities();
+    try {
+      maxNumber = Integer.parseInt(request.getParameter("max"));
+    } catch (NumberFormatException e) {}
+    
+    List<Entity> resultsList = results.asList(FetchOptions.Builder.withLimit(maxNumber));
+    List<String> myToDos = new ArrayList<>(); 
+
+    for (Entity entity : resultsList) {
+      name = (String) entity.getProperty("name");
+      category = (String) entity.getProperty("category");
+      content = (String) entity.getProperty("content");
+      comment = (String) entity.getProperty("comment");
+      String putTogether = name + "'s " + category + " recommendation: " + content+". ";
+      putTogether += name + " commented, \"" + comment + "\"";
+      myToDos.add(putTogether);
+    }
+
+    String jsonResponse = convertToJson(myToDos);
     response.setContentType("application/json");
     response.getWriter().println(jsonResponse);
   }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Oh, just because in the walkthroughs for this week that how they guided us through it. But i also saw an example in the walkthrough where get and post and delete are in three different servlet paths. Which would you recommend?